core/domain/user_jobs_one_off_test.py

```diff
@@ -842,3 +842,143 @@ def _mock_fetch_gravatar(unused_email):
+
+        self.assertIsNone(
+            user_services.get_user_settings(self.editor_id).
+            last_created_an_exploration)
+        self.assertIsNone(
+            user_services.get_user_settings(self.editor_id).
+            last_edited_an_exploration)
+
+        self._run_one_off_job()
+
+        self.assertIsNotNone(
+            user_services.get_user_settings(self.editor_id).
+            last_edited_an_exploration)
+        self.assertIsNone(
+            user_services.get_user_settings(self.editor_id).
+            last_created_an_exploration)
+
+    def test_that_last_edited_and_created_time_both_updated(self):
+        self.login(self.OWNER_EMAIL)
+        self.save_new_valid_exploration(
+            self.exp_id, self.owner_id, end_state_name='End')
+        exp_services.update_exploration(self.owner_id, self.exp_id, [{
+            'cmd': 'edit_exploration_property',
+            'property_name': 'objective',
+            'new_value': 'the objective'
+        }], 'Test edit')
+        self.logout()
+        self.login(self.EDITOR_EMAIL)
+        exp_services.update_exploration(self.editor_id, self.exp_id, [{
+            'cmd': 'edit_exploration_property',
+            'property_name': 'objective',
+            'new_value': 'new objective'
+        }], 'Test edit new')
+        self.logout()
+
+        user_settings = user_services.get_user_settings(self.owner_id)
+        user_settings.last_created_an_exploration = None
+        user_settings.last_edited_an_exploration = None
+        user_services._save_user_settings(user_settings)  # pylint: disable=protected-access
+        user_settings = user_services.get_user_settings(self.editor_id)
+        user_settings.last_edited_an_exploration = None
+        user_services._save_user_settings(user_settings)  # pylint: disable=protected-access
+
+        self.assertIsNone(
+            user_services.get_user_settings(self.owner_id).
+            last_created_an_exploration)
+        self.assertIsNone(
+            user_services.get_user_settings(self.owner_id).
+            last_edited_an_exploration)
+        self.assertIsNone(
+            user_services.get_user_settings(self.editor_id).
+            last_created_an_exploration)
+        self.assertIsNone(
+            user_services.get_user_settings(self.editor_id).
+            last_edited_an_exploration)
+
+        self._run_one_off_job()
+
+        self.assertIsNotNone(
+            user_services.get_user_settings(self.owner_id).
+            last_edited_an_exploration)
+        self.assertIsNotNone(
+            user_services.get_user_settings(self.owner_id).
+            last_created_an_exploration)
+        self.assertIsNotNone(
+            user_services.get_user_settings(self.editor_id).
+            last_edited_an_exploration)
+
+        self.assertIsNone(
+            user_services.get_user_settings(self.editor_id).
+            last_created_an_exploration)
```

Comment:
Good tests! I suggest adding one more for the "has not edited/created anything" case.